File: springboot_mp/src/main/java/com/xeon/springboot_mp/service/UserService.java
package com.xeon.springboot_mp.service;

import com.baomidou.mybatisplus.extension.service.IService;
import com.xeon.springboot_mp.dao.data.User;
import org.springframework.stereotype.Service;


/*
 * 固定写法 继承自IService，提供了常用的增删改查方法
 * */
public interface UserService extends IService<User> {
}

File: kotlin-springboot-kts/src/main/kotlin/com/yzq/kotlinspringbootkts/service/UserService.kt
package com.yzq.kotlinspringbootkts.service

import com.yzq.kotlinspringbootkts.mapper.UserMapper
import com.yzq.kotlinspringbootkts.data.User
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class UserService {
    @Autowired
    private lateinit var userMapper: UserMapper


    fun list(): List<User> {
        return userMapper.selectList(null)
    }

    fun getById(id: Long): User? {
        return userMapper.selectById(id)
    }

    fun create(user: User): Int {
        return userMapper.insert(user)
    }

    fun updateById(user: User): Int {
        return userMapper.updateById(user)
    }

    fun deleteById(id: Long): Int {
        return userMapper.deleteById(id)
    }
}
